### ISSUE TITLE
Discussion: customize loss function? 

### ISSUE THREAD

**Issue by patrickvonplaten** [patrickvonplaten/TRexGameRL#20]

**Comment by kilsenp** [comment]
Isn't it seen as a regression problem, so the network is trying to predict the reward? Therefore, a normalization would not be necessary.

**Comment by patrickvonplaten** [comment]
yeah true. The problem is that the losses become too big. I'm getting inf values pretty soon -> I think we should implement a customized loss function -> maybe compress the loss function with log()

**Comment by patrickvonplaten** [comment]
?

**Comment by kilsenp** [comment]
I had the same problem but after switching the optimizer to adam and using the default learning rate 1e-3, the loss went down (sometimes). I still need to log it over a longer time period.
But SGD should also work, maybe just dial down the learning rate?

On the other hand, would compressing it with log actually work? A small mean-squared-error will be go towards -inf and lead to big gradients, wouldn't it?

**Comment by patrickvonplaten** [comment]
maybe mean_squared_logarithmic_error is worth trying out

**Comment by patrickvonplaten** [comment]
mean_squared_logarithmic_error seems to work better for me

**Comment by patrickvonplaten** [comment]
mean_absolute_error might be a possibility as well

**Comment by patrickvonplaten** [comment]
https://stackoverflow.com/questions/47840527/using-tensorflow-huber-loss-in-keras -> huber loss or log cosh seems to be the most appropriate for RL
